ShopAppTests/App/Account/Controllers/SettingsViewControllerSpec.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  SettingsViewControllerSpec.swift
+//  ShopAppTests
+//
+//  Created by Radyslav Krechet on 3/1/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Nimble
+import Quick
+
+@testable import ShopApp
+
+class SettingsViewControllerSpec: QuickSpec {
+    override func spec() {
+        var viewController: SettingsViewController!
+        var viewModelMock: SettingsViewModelMock!
+        var tableProvider: SettingsTableProvider!
+        var tableView: UITableView!
+        
+        beforeEach {
+            viewController = UIStoryboard(name: StoryboardNames.account, bundle: nil).instantiateViewController(withIdentifier: ControllerIdentifiers.settings) as! SettingsViewController
+            
+            let navigationController = NavigationController(rootViewController: UIViewController())
+            navigationController.pushViewController(viewController, animated: false)
+            
+            let repositoryMock = AuthentificationRepositoryMock()
+            let updateCustomerUseCaseMock = UpdateCustomerUseCaseMock(repository: repositoryMock)
+            let loginUseCaseMock = LoginUseCaseMock(repository: repositoryMock)
+            let customerUseCaseMock = CustomerUseCaseMock(repository: repositoryMock)
+            viewModelMock = SettingsViewModelMock(updateCustomerUseCase: updateCustomerUseCaseMock, loginUseCase: loginUseCaseMock, customerUseCase: customerUseCaseMock)
+            viewController.viewModel = viewModelMock
+            
+            tableProvider = SettingsTableProvider()
+            viewController.tableProvider = tableProvider
+            
+            tableView = self.findView(withAccessibilityLabel: "tableView", in: viewController.view) as! UITableView
+        }
+        
+        describe("when view loaded") {
+            it("should have a correct view model type") {
+                expect(viewController.viewModel).to(beAKindOf(SettingsViewModel.self))
+            }
+            
+            it("should have correct title") {
+                expect(viewController.title) == "ControllerTitle.Settings".localizable
+            }
+            
+            it("should have correct back button image") {
```

Comment:
I can't find code where back button is set

Reply:
There is a setup in base view controller, but it will be different result in all subclasses (has back button or not) so we need to check it in all subclasses.